code/harness/DiskContents.cpp

```diff
@@ -377,7 +359,6 @@ bool DiskContents::compare_entries_at_path(DiskContents &compare_disk,
     }
   }
 
-  // TODO(P.S.) Fix the unmount issue and uncomment the function below.
   compare_disk.unmount_and_delete_mount_point();
   return retValue;
 }
```

Comment:
nit: `compare_file_contents()` may want to compare segments of the file at a time up to some maximum buffer size. For now, this works, but if we were comparing say, 2 1GB files on a system with 2GB of memory, it's quite likely that we'd run out of memory

Reply:
nit: in the future, we may want to either make `delete_files` recursive, or be careful to call `sanity_checks` starting from nested directories so that `delete_files` won't fail on directories that have non-empty directories nested in them

Reply:
deleteFiles is recursive and all the testFiles we create are under the single folder /mnt/snapshot right; So I guess it should work?